src/component/plotly-dash-preview/render.js

```diff
@@ -0,0 +1,62 @@
+const remote = require('../../util/remote')
+
+/**
+ * @param {object} info : info object
+ *  - layoutType TODO
+ *  - direction TODO
+ *  - backgroundColor
+ *  - panels
+ * @param {object} opts : component options
+ * @param {function} sendToMain
+ *  - errorCode
+ *  - result
+ *    - imgData
+ */
+function render (info, opts, sendToMain) {
+  const PRINT_TO_PDF = (info.format === 'pdf')
+  const result = {}
+
+  let win = remote.createBrowserWindow({
+    width: info.width,
+    height: info.height
+  })
+
+  const contents = win.webContents
+  let errorCode = null
+
+  const done = errorCode => {
+    win.close()
+
+    if (errorCode) {
+      result.msg = 'dash preview generation failed'
+    }
+    sendToMain(errorCode, result)
+  }
+
+  win.on('closed', () => {
+    win = null
+  })
+
+  contents.on('page-favicon-updated', () => {
+    if (PRINT_TO_PDF) {
+      contents.printToPDF({}, (err, pdfData) => {
+        if (err) {
+            done(525)
+        }
+        else {
+          result.imgData = pdfData
+          done()
+        }
+      })
+    } else {
+      contents.capturePage(img => {
+        result.imgData = img.toPNG()
```

Comment:
interesting, i'm curious what a PNG looks like! this could be very cool for thumbnails 😸

Reply:
Yes, that's the plan. I did test it with the vanguard app:

![a](https://user-images.githubusercontent.com/1546532/37477086-cde8e7f8-289c-11e8-8349-26661e0f4168.png)

It only captures the visible area in the browser, hence the need for `width`/`height` options ..